zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/approval/ApproveTask.java

```diff
@@ -329,18 +323,77 @@ private static ContentDetailResolver getDefaultContentDetailResolver() {
+            throw new IllegalArgumentException("approval task unsuccessful: collection.description required but was " +
                     "null");
         }
         if (session == null) {
-            throw new IllegalArgumentException("approval task unsuccesful: as session required but was null");
+            throw new IllegalArgumentException("approval task unsuccessful: as session required but was null");
         }
         if (StringUtils.isEmpty(session.getEmail())) {
-            throw new IllegalArgumentException("approval task unsuccesful: as session.email required but was null/empty");
+            throw new IllegalArgumentException(
+                    "approval task unsuccessful: as session.email required but was null/empty");
         }
-        info().data("collectionId", collection.getDescription().getId()).log("approval task: validation sucessful");
+        info().data("collectionId", collection.getDescription().getId()).log("approval task: validation successful");
+    }
+
+    protected void uploadNewEndpoint(Collection collection, CollectionReader collectionReader) throws ZebedeeException, IOException {
+        if (Configuration.isUploadNewEndpointEnabled()) {
+            uploadWhitelistedFiles(collection, collectionReader);
+        }
+    }
+    
+    protected void uploadWhitelistedFiles(Collection collection, CollectionReader collectionReader) throws ZebedeeException, IOException {
+        for (String uri : collectionReader.getReviewed().listUris()) {
+            if (uri.endsWith(".csv") || uri.endsWith(".xlsx")) {
+                String fileName = uri.substring(1);
+                Resource myFile = collectionReader.getResource(fileName);
+                if (DatasetWhitelistChecker.isWhitelisted(myFile.getName())) {
+                    info().log("File is whitelisted");
+                    uploadFile(myFile, fileName, collection.getDescription().getId());
+                } else {
+                    info().log("File is not whitelisted");
+                }
+                break;
+            }
+        }
+    }
+    
+    protected void uploadFile(Resource myFile, String fileName, String collectionId) throws ZebedeeException, IOException {
+        File file = new File("afile");
+        try (FileOutputStream outputStream = new FileOutputStream(file)) {
+            IOUtils.copy(myFile.getData(), outputStream);
+        } catch (FileNotFoundException e) {
+            info().log("file not found");
+            throw e;
+        } catch (IOException e) {
+            info().log("input/output error");
+            throw e;
+        }
+        List<NameValuePair> params = createUploadParams(fileName, "path", collectionId);
+        String uploadServiceURL = Configuration.getUploadServiceApiUrl() + "upload-new";
+        Client uploadServiceClient = new APIClient(uploadServiceURL, Configuration.getServiceAuthToken());
```

Comment:
Existing patterns within zebedee should be followed for implementing the client which should be done via an interface and the implementation of the service.

Example of how the static-files-client-java is implemented can be found here - https://github.com/search?q=repo%3AONSdigital%2Fzebedee+StaticFilesService&type=code

Reply:
This is now addressed.